core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.spec.ts

```diff
@@ -657,59 +670,62 @@ describe('Learner dashboard page', function() {
-
-      expect(ctrl.messageSendingInProgress).toBe(true);
-
-      $httpBackend.flush();
-      expect(ctrl.messageSendingInProgress).toBe(false);
-    });
+    it('should add a new message in a thread when thread is selected',
+      function() {
+        var threadStatus = 'open';
+        var explorationId = 'exp1';
+        var threadId = 'thread_1';
+        var explorationTitle = 'Exploration Title';
+        var threadMessages = [{
+          message_id: '1',
+          text: 'Feedback 1',
+          updated_status: 'open',
+          suggestion_html: 'An instead of a',
+          current_content_html: 'A orange',
+          description: 'Suggestion for english grammar',
+          author_username: 'username2',
+          author_picture_data_url: 'foo',
+          created_on_msecs: 1200
+        }];
+
+        $httpBackend.expect('GET', '/learnerdashboardthreadhandler/thread_1')
+          .respond({
+            message_summary_list: threadMessages
+          });
+        ctrl.onClickThread(
+          threadStatus, explorationId, threadId, explorationTitle);
+        $httpBackend.flush();
+
+        var threadId = 'thread_1';
+        var message = 'This is a new message';
+        $httpBackend.expect('POST', '/threadhandler/' + threadId).respond(200);
+        ctrl.addNewMessage(threadId, message);
+
+        expect(ctrl.messageSendingInProgress).toBe(true);
+
+        $httpBackend.flush();
+        expect(ctrl.messageSendingInProgress).toBe(false);
+      });
 
-    it('should show suggestion modal', function() {
-      spyOn(SuggestionModalForLearnerDashboardService, 'showSuggestionModal')
-        .and.callThrough();
-      var newContent = 'New content';
-      var oldContent = 'Old content';
-      var description = 'Description';
-      ctrl.showSuggestionModal(newContent, oldContent, description);
-
-      expect(SuggestionModalForLearnerDashboardService.showSuggestionModal)
-        .toHaveBeenCalledWith('edit_exploration_state_content', {
-          newContent: newContent,
-          oldContent: oldContent,
-          description: description
-        });
-    });
+    it('should show new and old content when opening suggestion modal',
+      function() {
+        spyOn(SuggestionModalForLearnerDashboardService, 'showSuggestionModal')
+          .and.callThrough();
+        var newContent = 'New content';
+        var oldContent = 'Old content';
+        var description = 'Description';
+        ctrl.showSuggestionModal(newContent, oldContent, description);
+
+        expect(SuggestionModalForLearnerDashboardService.showSuggestionModal)
+          .toHaveBeenCalledWith('edit_exploration_state_content', {
+            newContent: newContent,
+            oldContent: oldContent,
+            description: description
+          });
+      });
 
-    it('should open uib modal when removing activity', function() {
+    it('should open remove activity modal with $uibModal when removing' +
```

Comment:
```suggestion
    it('should open remove activity modal when removing' +
```
That is an unnecessary code detail

Reply:
Done.